typescript/packages/planning-server/src/index.ts

```diff
@@ -1,71 +1,197 @@
 import { serve } from "https://deno.land/std@0.140.0/http/server.ts";
-import { streamText } from "npm:ai";
+import { streamText, generateText } from "npm:ai";
 import { ensureDir } from "https://deno.land/std/fs/mod.ts";
 import { crypto } from "https://deno.land/std/crypto/mod.ts";
 import { anthropic } from "npm:@ai-sdk/anthropic";
 import { config } from "https://deno.land/x/dotenv/mod.ts";
 import { groq } from "npm:@ai-sdk/groq";
+import { openai } from "npm:@ai-sdk/openai";
+import { vertex } from "npm:@ai-sdk/google-vertex";
 
 await config({ export: true });
 
 const CACHE_DIR = "./cache";
 
+const MODELS: Record<
+  string,
+  { model: any; contextWindow: number; maxOutputTokens: number }
+> = {
+  "anthropic:claude-3.5-haiku": {
+    model: anthropic("claude-3-5-haiku-20241022"),
+    contextWindow: 200000,
+    maxOutputTokens: 8192,
+  },
+  "anthropic:claude-3.5-sonnet": {
+    model: anthropic("claude-3-5-sonnet-20241022"),
+    contextWindow: 200000,
+    maxOutputTokens: 8192,
+  },
+  "anthropic:claude-3-opus": {
+    model: anthropic("claude-3-opus-20240229"),
+    contextWindow: 200000,
+    maxOutputTokens: 4096,
+  },
+  "groq:llama-3.1-70b": {
+    model: groq("llama-3.1-70b-versatile"),
+    contextWindow: 128000,
+    maxOutputTokens: 8000,
+  },
+  "groq:llama-3.2-11b-vision": {
+    model: groq("llama-3.2-11b-vision-preview"),
+    contextWindow: 128000,
+    maxOutputTokens: 8000,
+  },
+  "groq:llama-3.2-90b-vision": {
+    model: groq("llama-3.2-90b-vision-preview"),
+    contextWindow: 128000,
+    maxOutputTokens: 8000,
+  },
+  "openai:gpt-4o": {
+    // TODO
+    model: openai("gpt-4o"),
+    contextWindow: 128000,
+    maxOutputTokens: 16384,
+  },
+  "openai:gpt-4o-mini": {
+    // TODO
+    model: openai("gpt-4o-mini"),
+    contextWindow: 128000,
+    maxOutputTokens: 16384,
+  },
+  "openai:o1-preview": {
+    // TODO
+    model: openai("o1-preview-2024-09-12"),
+    contextWindow: 128000,
+    maxOutputTokens: 32768,
+  },
+  "openai:o1-mini": {
+    // TODO
+    model: openai("o1-mini-2024-09-12"),
+    contextWindow: 128000,
+    maxOutputTokens: 65536,
+  },
+  "google:gemini-1.5-flash": {
+    model: vertex("gemini-1.5-flash-002"),
+    contextWindow: 1000000,
+    maxOutputTokens: 8192,
+  },
+  "google:gemini-1.5-pro": {
+    model: vertex("gemini-1.5-pro-002"),
+    contextWindow: 1000000,
```

Comment:
we should only add models if the env keys are set... otherwise you get things like:

    error: Uncaught (in promise) AI_LoadSettingError: Google Vertex project setting is missing. Pass it using the 'project' parameter or the GOOGLE_VERTEX_PROJECT environment variable.
        at loadSetting (file:///Users/jesse/ct/labs/typescript/node_modules/@ai-sdk/google-vertex/node_modules/@ai-sdk/provider-utils/dist/index.mjs:150:11)
        at loadVertexProject (file:///Users/jesse/ct/labs/typescript/node_modules/@ai-sdk/google-vertex/dist/index.mjs:716:35)
        at createVertexAI (file:///Users/jesse/ct/labs/typescript/node_modules/@ai-sdk/google-vertex/dist/index.mjs:731:16)
        at createChatModel (file:///Users/jesse/ct/labs/typescript/node_modules/@ai-sdk/google-vertex/dist/index.mjs:740:17)
        at provider (file:///Users/jesse/ct/labs/typescript/node_modules/@ai-sdk/google-vertex/dist/index.mjs:759:12)
        at file:///Users/jesse/ct/labs/typescript/packages/planning-server/src/index.ts:74:12